### PR TITLE
test lit in a simple dropdown

### PR DESCRIPTION
**Pull request by TrygveUrdahl**
# PR Checklist
https://elvia.atlassian.net/wiki/spaces/TEAMATOM/pages/10427498683/Review+prosess

- [ ] Bumpet package?
- [ ] Updated changelog?
- [ ] Correct date in changelog?

### CODE REVIEW

File: packages/elvis-dropdown-lit/src/e-option.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  value: string;
+
+  @property({ type: Boolean })
+  disabled = false;
+
+  @consume({ context: dropdownContext, subscribe: true })
+  dropdownContext: DropdownContext;
+
+  render() {
+    const selected = this.dropdownContext.value === this.value;
```
Comment (TrygveUrdahl) [inline packages/elvis-dropdown-lit/src/e-option.ts:56]: I haven't gotten this `selected`-value to work yet, for some reason it doesn't update when the value in the context changes.

File: .eslintrc.json [hunk 2]
```diff
@@ -2,7 +2,6 @@
   "root": true,
   "plugins": ["@typescript-eslint"],
   "rules": {
-    "no-console": ["error", { "allow": ["warn", "clear", "info", "error", "dir", "trace", "table"] }],
```
Comment (eTallang) [inline .eslintrc.json:5]: Why are these removed? 🤓

File: packages/elvis-dropdown-lit/package.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  "main": "dist/index.js",
+  "types": "dist/index.d.ts",
+  "type": "module",
+  "files": [
+    "dist",
+    "react.ts"
+  ],
+  "scripts": {
+    "build": "tsup src/index.ts --dts --format esm --out-dir dist",
+    "start": "tsup src/index.ts --sourcemap --dts --format esm --out-dir dist --watch"
```
Comment (eTallang) [inline packages/elvis-dropdown-lit/package.json:13]: So clean and easy to read ❤️ And as I hoped (and hope we can keep moving forwards); no custom build. Could we do this with `esbuild` and skip the typings? Or are they intended for React users?

File: packages/elvis-dropdown-lit/package.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    "@elvia/elvis-typography": "^2.7.1",
+    "@lit/context": "^1.1.1",
+    "@lit/react": "^1.0.5",
+    "lit": "^3.1.3",
+    "lit-html": "^3.1.3"
+  },
+  "devDependencies": {
+    "tsup": "^8.0.2",
+    "typescript": "~5.4.5"
+  }
```
Comment (eTallang) [inline packages/elvis-dropdown-lit/package.json:27]: Should we have `react` as an optional dependency? Considering we do an import from `react` in our `react.ts` file

File: packages/elvis-dropdown-lit/package.json (hunk 3 above)
Comment (eTallang) [reply packages/elvis-dropdown-lit/package.json:13]: We must also think about what to do with the CEG again (considering we don't export any prop interfaces). Maybe there exists a way of extracting the props from the custom element and infer a type from it that we can feed into the existing types for the CEG?

File: packages/elvis-dropdown-lit/src/e-dropdown.ts [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+      --width: 448px;
+      --option-height: 48px;
+      display: block;
+      width: var(--width);
+      position: relative;
+    }
+
+    .input {
+      width: inherit;
+      border: 1px solid ${unsafeCSS(getThemeColor('border-1'))};
```
Comment (eTallang) [inline packages/elvis-dropdown-lit/src/e-dropdown.ts:25]: We could consider removing the need for the call to `unsafeCss` by using the CSS variables directly, in other words `var(--e-color-border-1)`.

File: packages/elvis-dropdown-lit/src/e-dropdown.ts [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+  }
+
+  disconnectedCallback(): void {
+    super.disconnectedCallback();
+    document.removeEventListener('click', this.closeOnOutsideClick);
+  }
+
+  private handleOpen() {
+    this.isOpen = true;
+  }
```
Comment (eTallang) [inline packages/elvis-dropdown-lit/src/e-dropdown.ts:135]: Tiny thing, but we could consider to use arrow-function syntax for this guy as well (or go for the "regular" function syntax, as it looks better when we utilize life cycle hooks like `connectedCallback()`)

File: packages/elvis-dropdown-lit/src/e-dropdown.ts [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+
+  private triggerValueChangeEvent(value: string) {
+    this.dispatchEvent(new CustomEvent('valueChange', { detail: { value } }));
+  }
+}
+
+declare global {
+  interface HTMLElementTagNameMap {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
```
Comment (eTallang) [inline packages/elvis-dropdown-lit/src/e-dropdown.ts:166]: 😆

File: packages/elvis-dropdown-lit/src/e-option.ts [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+@customElement('e-option')
+export class EOption extends LitElement {
+  static styles = css`
+    :host {
+      height: var(--option-height);
+      display: flex;
+      align-items: center;
+
+      ${unsafeCSS(getTypographyCss('text-md'))};
```
Comment (eTallang) [inline packages/elvis-dropdown-lit/src/e-option.ts:18]: This makes sense. It would be very cool if we could build these styles into the component through a "build" so that we could have `@elvia/elvis-typography` as a dev dependency

File: packages/react-test/src/App.tsx [hunk 9]
```diff
@@ -202,21 +203,22 @@ function App() {
...
+                <EDropdownComponent
+                  label="Label"
+                  placeholder="Plassholder"
+                  onValueChange={(e) => {
+                    console.log('Value changed:', e.detail.value);
+                  }}
+                >
+                  {[...Array(10)].map((_, i) => (
+                    <EOptionComponent key={i} value={i.toString()}>
+                      <div className="e-flex e-gap-8 e-w-100" style={{ justifyContent: 'space-between' }}>
```
Comment (eTallang) [inline packages/react-test/src/App.tsx:215]: Tiny thing, but this is probably something we could do through the styles on the `EOptionComponent`

File: .eslintrc.json (hunk 2 above)
Comment (TrygveUrdahl) [reply .eslintrc.json:5]: Honestly, just because I was tired of the warnings while I was debugging 😂

File: packages/elvis-dropdown-lit/package.json (hunk 3 above)
Comment (TrygveUrdahl) [reply packages/elvis-dropdown-lit/package.json:13]: I agree, super nice and clean!
I believe the typings are used for React, and also for anyone who does a `document.querySelector('')` or even when just looking at the type of the custom element tag in Angular. 
image
Here I am just hovering the element, and it sees what type it has! This is due to the 
`declare global {
  interface HTMLElementTagNameMap {
    'e-dropdown': EDropdown;
  }
}`

File: packages/elvis-dropdown-lit/src/e-dropdown.ts (hunk 5 above)
Comment (TrygveUrdahl) [reply packages/elvis-dropdown-lit/src/e-dropdown.ts:25]: While I kind of agree, I also think it's nice that this includes the fallback color-value, in case the component is used
1. without @elvia/elvis
2. with an unsupported version of @elvia/elvis, since there is no explicit version required in the `package.json`

File: packages/elvis-dropdown-lit/src/e-dropdown.ts (hunk 7 above)
Comment (TrygveUrdahl) [reply packages/elvis-dropdown-lit/src/e-dropdown.ts:166]: Yeah, Typescript was complaining because this was being injected into the `global interface HTMLElementTagNameMap` both here _and_ in the built dist-files, so this was the easiest quick-fix I could thing of 🥴

File: packages/elvis-dropdown-lit/package.json (hunk 3 above)
Comment (TrygveUrdahl) [reply packages/elvis-dropdown-lit/package.json:13]: We should be able to access the prop interfaces from the React-component through something like `type Props = PropsWithoutRef<typeof EDropdownComponent>` 🤔
